x/superfluid/keeper/stake_test.go

```diff
@@ -148,6 +151,112 @@ func (suite *KeeperTestSuite) TestSuperfluidDelegate() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestValidateLockForSFDelegate() {
+	lockOwner := suite.TestAccs[0]
+
+	tests := []struct {
+		name                             string
+		lock                             *lockuptypes.PeriodLock
+		sender                           string
+		skParams                         types.Params
+		superfluidAssetToSet             types.SuperfluidAsset
+		lockIdAlreadySuperfluidDelegated bool
+		expectedErr                      error
+	}{
+		{
+			name: "valid gamm lock",
+			lock: &lockuptypes.PeriodLock{
+				Owner:    lockOwner.String(),
+				Coins:    sdk.NewCoins(sdk.NewCoin("gamm/pool/1", sdk.NewInt(100))),
+				Duration: time.Hour * 24 * 21,
+				ID:       1,
+			},
+			superfluidAssetToSet: types.SuperfluidAsset{Denom: "gamm/pool/1", AssetType: types.SuperfluidAssetTypeLPShare},
+			expectedErr:          nil,
+		},
+		{
+			name: "valid cl lock (suffix properly removed)",
```

Comment:
What does "suffix properly removed" mean?

Reply:
When we run validation on cl denoms, the validation logic strips the position ID suffix and compares the remaining to approved SF denoms. Extended the test name here https://github.com/osmosis-labs/osmosis/pull/4883/commits/cb732b03958b21a7efa940a99cb6ab0ac531bc0d